### PR TITLE
feat: add encumber aggregate utxo creation method for wallet cli

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Description
---
Given a list of signatures, and other metadata, it is desirable to create an encumber aggregate utxo. This PR attempts to produce logic for this new feature.

Motivation and Context
---

How Has This Been Tested?
---

### CODE REVIEW

File: base_layer/core/src/transactions/transaction_components/unblinded_output.rs [hunk 1]
```diff
@@ -201,6 +201,51 @@ impl UnblindedOutput {
         ))
     }
 
+    pub fn as_transaction_input_with_partial_signature(
+        &self,
+        factory: &CommitmentFactory,
+        script_public_key: PublicKey,
```
Comment (SWvheerden) [inline base_layer/core/src/transactions/transaction_components/unblinded_output.rs]:
```suggestion
        partial_script_public_key: PublicKey,
```

File: base_layer/core/src/transactions/transaction_components/unblinded_output.rs [hunk 2]
```diff
@@ -201,6 +201,51 @@ impl UnblindedOutput {
         ))
     }
 
+    pub fn as_transaction_input_with_partial_signature(
```
Comment (SWvheerden) [inline base_layer/core/src/transactions/transaction_components/unblinded_output.rs:206]:
```suggestion
///
  add docs here as this is a public function, make sure to mention that the two pubkeys exclude the callers privates key
  ///
    pub fn as_transaction_input_with_partial_signature(
```

File: base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs [hunk 3]
```diff
@@ -476,7 +476,7 @@ impl SenderTransactionInitializer {
         self.amounts.clone().into_vec().iter().sum()
     }
 
-    pub(super) fn fee(&self) -> &Fee {
+    pub fn fee(&self) -> &Fee {
```
Comment (SWvheerden) [inline base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs]:
```suggestion
    pub(super) fn fee(&self) -> &Fee {
```

leave this as is

File: base_layer/p2p/src/services/liveness/service.rs [hunk 4]
```diff
@@ -161,7 +161,7 @@ where
         match ping_pong_msg.kind().ok_or(LivenessError::InvalidPingPongType)? {
             PingPong::Ping => {
                 self.state.inc_pings_received();
-                self.send_pong(ping_pong_msg.nonce, public_key).await.unwrap();
+                self.send_pong(ping_pong_msg.nonce, public_key).await?;
```
Comment (SWvheerden) [inline base_layer/p2p/src/services/liveness/service.rs]: Why this change?

Comment (SWvheerden) [reply base_layer/p2p/src/services/liveness/service.rs]: A PR should fix one thing only

File: base_layer/wallet/src/output_manager_service/service.rs [hunk 5]
```diff
@@ -1231,6 +1260,138 @@ where
...
+            .with_input(
+                input.clone().as_transaction_input_with_partial_signature(
+                    &self.resources.factories.commitment,
+                    total_script_pubkey,
+                    total_signature_nonce,
+                )?,
+                input.clone(),
+            );
+
+        let fee = builder.fee().calculate(fee_per_gram, 1, 1, 1, metadata_byte_size);
```
Comment (SWvheerden) [inline base_layer/wallet/src/output_manager_service/service.rs]: this fee comes from the consensus constants, you don't have to expose the builders one, just create a new one here

File: base_layer/wallet/src/transaction_service/service.rs [hunk 6]
```diff
@@ -986,6 +1029,240 @@ where
...
+        info!(
+            target: LOG_TARGET,
+            "Finalized create n of m transaction TxId: {}", tx_id
+        );
+
+        // This event being sent is important, but not critical to the protocol being successful. Send only fails if
+        // there are no subscribers.
+        let _size = self
+            .event_publisher
+            .send(Arc::new(TransactionEvent::TransactionCompletedImmediately(tx_id)));
```
Comment (SWvheerden) [inline base_layer/wallet/src/transaction_service/service.rs]: We dont want to complete this yet, remember this is incomplete

File: base_layer/wallet/src/transaction_service/service.rs [hunk 7]
```diff
@@ -986,6 +1029,240 @@ where
...
+                tx.clone(),
+                TransactionStatus::Completed,
+                "".to_string(),
+                Utc::now().naive_utc(),
+                TransactionDirection::Outbound,
+                None,
+                None,
+                None,
+            ),
+        )?;
```
Comment (SWvheerden) [inline base_layer/wallet/src/transaction_service/service.rs]: We dont want to complete this yet, remember this is incomplete
leave this as pending

File: base_layer/p2p/src/services/liveness/service.rs (hunk 4 above)
Comment (jorgeantonio21) [reply base_layer/p2p/src/services/liveness/service.rs]: Somehow this was changed, I don't remember having looked at this file. Let me rewind the code to the previous line of code.

File: base_layer/core/src/transactions/transaction_components/unblinded_output.rs (hunk 1 above)
Comment (jorgeantonio21) [reply base_layer/core/src/transactions/transaction_components/unblinded_output.rs]: Perfect :)

File: base_layer/core/src/transactions/transaction_components/unblinded_output.rs (hunk 2 above)
Comment (jorgeantonio21) [reply base_layer/core/src/transactions/transaction_components/unblinded_output.rs:206]: Added documentation as requested !

File: base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs (hunk 3 above)
Comment (jorgeantonio21) [reply base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs:479]: How can I access this method from the `base_layer/wallet` folder ?

File: base_layer/p2p/src/services/liveness/service.rs (hunk 4 above)
Comment (jorgeantonio21) [reply base_layer/p2p/src/services/liveness/service.rs]: Actually this change comes from #4784. I will change it back to unwrap, and when we merge to `development` we can resolve the conflict.

File: base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs (hunk 3 above)
Comment (jorgeantonio21) [reply base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs]: Done !

File: base_layer/wallet/src/output_manager_service/service.rs (hunk 5 above)
Comment (jorgeantonio21) [reply base_layer/wallet/src/output_manager_service/service.rs]: Refactored accordingly :)

**Review by stringhandler — APPROVED** [review]
